libmscore/layout.cpp

```diff
@@ -3448,8 +3448,15 @@ System* Score::collectSystem(LayoutContext& lc)
             for (Element* e : s->annotations()) {
                   if (e->isStaffText() || e->isSystemText() || e->isInstrumentChange())
                         e->layout();
-                  if (e->isHarmony())
+                  if (e->isHarmony()) {
+                        Harmony* h = toHarmony(e);
+                        // Layout of harmony seems to be bound currently
+                        // to ChordRest (see ChordRest::shape). But harmony
+                        // can exist without chord or rest too.
+                        if (h->isLayoutInvalid())
```

Comment:
@dmitrio95 Do you remember why the check for isLayoutInvalid() here?  Was it just performance (to avoid relayout of chord symbols that *are* attached to ChordRests) or something to do with the other hack about checking layoutvalid within the TextBase code?

I ask because other layout bugs with chord symbols are causing me want to skip this check here, but not if it breaks something else.  There might be other solutions, though.

Reply:
I put this check just because I did not want to change something about chord symbols attached to ChordRests as their layout was intentionally moved from here previously. Maybe it would be no harm in doing their re-layout here.